### DISCUSSION TITLE
[SSE Subscriptions] What's a good timeout configuration?

### DISCUSSION THREAD

**Discussion by cabello** [trpc/trpc#6294] [Q&A]
Thank you for bringing trpc to life, working on v11 and maintaining it, very grateful. 🙇 

From the [docs](https://trpc.io/docs/client/links/httpSubscriptionLink#server-ping), I am not sure what are the implications of `ping`, `maxDurationMs` and `reconnectAfterInactivityMs`.

ChatGPT suggested:
```ts
import { initTRPC } from '@trpc/server';
export const t = initTRPC.create({
  sse: {
    // Maximum duration of a single SSE connection in milliseconds
    // Reasonable value: 1 hour (3600000 ms) to 24 hours depending on server and use case
    maxDurationMs: 3600_000, 

    ping: {
      // Enable periodic ping messages to keep connection alive
      enabled: true,
      // Send ping message every 15-30 seconds
      // 2 seconds may be excessive unless you require high-frequency checks
      intervalMs: 15_000, 
    },

    client: {
      // Time the client should wait before attempting to reconnect after inactivity
      // Reasonable value: 2-5 seconds for responsive reconnection
      reconnectAfterInactivityMs: 5_000,
    },
  },
});
```

`maxDurationMs`, could that be taxing on my server? do I need CPU/RAM for each N connections?

I also saw in console log "no messages for 45 seconds, reconnecting" and I suspect this is the default for `reconnectAfterInactivityMs` so not sure if a `ping` being greater than `reconnectAfterInactivityMs` is conflicting or if `reconnectAfterInactivityMs` is really about going offline/online.

**Comment by KATT** [comment]
Thanks for your kind words.


Depends on your app - we would love feedback of what a useful default would be (right now we have no default, different browsers do it differently).

I don't know what a good rule of thumb is, but some common sense tells me something like: if you know your events should emit several times a second, set a really low `reconnectAfterInactivityMs`, if it's several events a minute, pick a higher one.

**Comment by testerez** [comment]
It's unclear to me what `maxDurationMs` is actually doing. I was expecting the `signal` passed to the procedure to be abort after that delay, but I can't see any effect of setting `maxDurationMs`...

More generally, since I'm hosted on Vercel, I'm trying to figure out the right way to reconnect periodically so that I don't hit the lambda timeout.
